lib/blocks.php

```diff
@@ -504,3 +504,94 @@ function gutenberg_migrate_old_typography_shape( $metadata ) {
 if ( ! function_exists( 'wp_migrate_old_typography_shape' ) ) {
 	add_filter( 'block_type_metadata', 'gutenberg_migrate_old_typography_shape' );
 }
+
+/**
+ * Enqueue a stylesheet for a specific block.
+ *
+ * If the theme has opted-in to separate-styles loading,
+ * then the stylesheet will be enqueued on-render,
+ * otherwise when the block inits.
+ *
+ * @param string $block_name The block-name, including namespace.
+ * @param array  $args       An array of arguments [handle,src,deps,ver,media].
+ * @param bool   $register   Whether the stylesheet should be registered.
+ *                           If set to false, then assume it has already been registered,
+ *                           and only enqueue it.
+ *
+ * @return void
+ */
+function gutenberg_enqueue_block_style( $block_name, $args, $register = true ) {
+	$args = wp_parse_args(
+		$args,
+		array(
+			'handle' => '',
+			'src'    => '',
+			'deps'   => array(),
+			'ver'    => false,
+			'media'  => 'all',
+		)
+	);
+
+	/**
+	 * Callback function to register and enqueue styles.
+	 *
+	 * @param string $content When the callback is used for the render_block filter,
+	 *                        the content needs to be returned so the function parameter
+	 *                        is to ensure the content exists.
+	 *
+	 * @return string
+	 */
+	$callback = function( $content ) use ( $args, $register ) {
+		// Register the stylesheet.
+		if ( $register ) {
+			wp_register_style( $args['handle'], $args['src'], $args['deps'], $args['ver'], $args['media'] );
+		}
+
+		// Add `path` data if provided.
+		if ( isset( $args['path'] ) ) {
+			wp_style_add_data( $args['handle'], 'path', $args['path'] );
+		}
+
+		// Enqueue the stylesheet.
+		wp_enqueue_style( $args['handle'] );
+
+		return $content;
+	};
+
+	$hook = did_action( 'wp_enqueue_scripts' ) ? 'wp_footer' : 'wp_enqueue_scripts';
+	if ( wp_should_load_separate_core_block_assets() ) {
+		$hook = "render_block_$block_name";
+	}
+
+	// Enqueue assets in the frontend.
+	add_filter( $hook, $callback );
+
+	// Enqueue assets in the editor.
+	add_action( 'enqueue_block_editor_assets', $callback );
+}
+
+/**
+ * Allow multiple block styles.
+ *
+ * @param array $metadata Metadata for registering a block type.
+ *
+ * @return array
+ */
+function gutenberg_multiple_block_styles( $metadata ) {
+	foreach ( array( 'style', 'editorStyle' ) as $key ) {
+		if ( isset( $metadata[ $key ] ) && is_array( $metadata[ $key ] ) ) {
+			foreach ( $metadata[ $key ] as $handle ) {
+				gutenberg_enqueue_block_style(
+					$metadata['name'],
+					array( 'handle' => $handle ),
+					false
+				);
+			}
+
+			// Only return the 1st item in the array.
+			$metadata[ $key ] = $metadata[ $key ][0];
```

Comment:
Shouldn't we shift the first element of the array and pass it as the item in the metadata and register all other styles with `gutenberg_enqueue_block_style`? I guess for the first element it should work just fine with the default metadata handling.

Reply:
Huh... using `array_shift` didn't cross my mind. Done :+1: